webview-ui/test-generation/src/App.tsx

```diff
@@ -115,26 +122,45 @@ function App() {
     status,
     steps,
     tunnel,
+    apps,
   } = state;
 
   return (
     <main>
       <section className="inputs">
         <h2>What do you want to test?</h2>
-        <VSCodeTextField
-          value={appName}
-          placeholder="From Sauce Labs App Storage, e.g. test.apk"
-          onInput={(e) => {
-            if (e.target && 'value' in e.target) {
-              dispatch({
-                type: 'setAppName',
-                value: e.target.value as string,
-              });
-            }
-          }}
-        >
-          Application Name
-        </VSCodeTextField>
+        <section className="with-label">
+          <label>App Name</label>
+          <VSCodeDropdown
+            onInput={(e) => {
+              if (e.target && 'value' in e.target) {
+                const appName: string = e.target.value as string;
+                const appInfo = apps.find(
+                  (app) => app.name === appName,
+                ) as AppInfo;
+                dispatch({
+                  type: 'setAppName',
+                  value: {
+                    appName: appName as string,
+                    platformName: appInfo.kind as 'iOS' | 'Android',
```

Comment:
We are still forcing types onto the compiler though 😅 If there's a bug in https://github.com/saucelabs/vscode-scriptiq/blob/7b7798be8bf82ec139a3453a1fb8ac0423c24560/webview-ui/test-generation/src/state.ts#L377-L377, this line will be complicit in the deception of the compiler.

**Options**
1. One technique that we use in saucectl quite often, is to massage the data that we receive at the HTTP client level, which would be in [fetchApps()](https://github.com/saucelabs/vscode-scriptiq/blob/7b7798be8bf82ec139a3453a1fb8ac0423c24560/src/api/llm/http.ts#L76-L76) in this case. You could return a type of AppInfo that's internal to us, with `platformName` being a native field of type `Platform`.
2. Add a function for type guarding. `AppKindToPlatform()` or some such. Might result in more 🍝 code over time though.